MyLinkedList/LinkedListClass.cs

```diff
@@ -1,8 +1,764 @@
 ﻿using System;
+using System.Text;
+using System.Collections;
+using System.Collections.Generic;
 
 namespace MyLinkedList
 {
-    public class MyLinkedList<T>
+    public class LinkedListClass<T> : IList<T>, IEnumerable<T> where T : IComparable
     {
+        private Node<T> _head;
+
+        private Node<T> _tail;
+
+        public int Count { get; set; }
+
+        public T this[int index]
+        {
+            get
+            {
+                if (IsValidIndex(index))
+                {
+                    Node<T> current = _head;
+
+                    for (int i = 1; i <= index; i++)
+                    {
+                        current = current.Next;
+                    }
+
+                    return current.Data;
+                }
+
+                throw new IndexOutOfRangeException("Invalid index!");
+            }
+            set
+            {
+                if (IsValidIndex(index) && !(value is null))
```

Comment:
`if(value is not null)`
`{`
`if(IsValidIndex(index))`
`{`
`}`
else
`{}`
`}`
`else`
`exception`